Topics/Development_Process/GraphQL_VS_REST.md

```diff
@@ -0,0 +1,94 @@
+
+### Designing APIs
+In this article, we will specifically discuss APIs that deal with sending requests over the internet, which is a very special case of APIs. We will cover two special commonly used types of APIs:
+1. REST
+2. GraphQL 
+
+On this page, you will learn about the differences between the two, and hopefully be able to make a confident decision on which one you need to use, based on your project requirements. 
+
+## REST Overview
+A REST API is any API that follows the REST architecture. The implementation of it can be handled in many pre-existing libraries, so this page will focus on the important details:
+1. In REST, you have to specify the `type` of the request you are sending when you call an API endpoint.
+    + Some common examples include `GET`, `POST`, `PUT`, and `DELETE`. Others exist, but these are the ones used most often.
+2. REST is not specific; when you call an endpoint, you will get *all* of the data that endpoint can give, whether you like it or not.
+    + This can be ideal when you require a lot of data, but can also lead to larger than necessary load on your system if you have to call many API endpoints in quick succession.
+3. Every difference type of request in REST has to be called to a different endpoint. For instance, if you want to `GET` all `Users` and `Cars` at the same time, with a typical REST API, you would need to send two seprate `GET` requests to a `/users` and `/cars` endpoint.
+
+
+## GraphQL Overview
+GraphQL is difference from REST in the sense that it is not actually an API architecture. In fact, it is actually a `query language`, hence the QL in its name. This leads to a number of distinct characteristics that we will discuss:
+1. In GraphQL there are only two types of requests: `Queries`, and `Mutations`. A query is used to get data, i.e, "query" some data. Everything else would be a mutation; creating, updating, or deleting data all mutate, and thus they are mutations.
+    + To follow proper GraphQL standards, any endpoint that only sends back data would fall under `query`. Any endpoint that changes data in any way would fall under `mutation`.
+2. GraphQL is very specific; when you call an endpoint, you can request *exactly* what data you want, and you can send different structures of data, based on customizable `types`.
+3. GraphQL only requires one endpoint. Usually, this is found at the `/graphql` URL on a webapp. Since GraphQL is a query language, whenever call the endpoint, you use the request body to specify your query. From this, you can define mutations and queries, and can in fact, get multiple pieces on data through one request. Using the same example from REST, you would only need to send one `/graphql` request, and just write both queries for data in the same request. This allows you to get the same amount of data using one request instead of two separate ones.

```

Comment:
I think you should clarify more on what you mean by sending both queries in the same request, perhaps with an example of some sort.

Reply:
Sounds good! Do you think that just an example would be sufficient, or should I explain it more in words?

Reply:
I think both would be good to have together! It would make this point a lot easier to understand for those who are new to GraphQL.

Reply:
That's a good idea, I think that would definitely aid the reader in understanding. Now that I think about it, an both an example and an explanation might be too much in one place, so breaking them up is great!

Reply:
Great, I've added them both!!